framework/db/sqlite/QueryBuilder.php

```
@@ -58,59 +61,81 @@ class QueryBuilder extends \yii\db\QueryBuilder
+        if ($placeholders !== []) {
+            $source = 'VALUES (' . implode(', ', $placeholders) . ')';
+        } elseif ($insertColumns instanceof Query) {
+            // A WHERE clause resolves SQLite's parsing ambiguity between a JOIN and the UPSERT ON CONFLICT clause.
+            $selectSql = ltrim($values);
+
+            $source = <<<SQL
+            SELECT * FROM ({$selectSql}) WHERE TRUE
+            SQL;
+        } else {
+            $source = ltrim($values);
         }
```

PHPMD (2.15.0)

[error] The method upsert uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them. (undefined)

(ElseExpression)